stuff2sort/System.py

```diff
@@ -0,0 +1,1150 @@
+        final_linkcorrlist = []
+        for element in reshaped_linkcorrlist:
+            found = False
+            for i in range(0, len(final_linkcorrlist)):
+                if (
+                    final_linkcorrlist[i][0] == element[0]
+                    and final_linkcorrlist[i][1] == element[1]
+                ):
+                    found = True
+                    break
+            if found:
+                continue
+            final_linkcorrlist.append(element)
+        return (
+            sorted(final_linkcorrlist, key=lambda l: l[0]),
+            q1list,
+            q2list,
+            q3list,
+            m3list,
+        )
+    
+    def getincludelist(self, topology_input):
+
+        '''
+        ------------------------------ \\
+        EFFECT: \\
+        ---------------  \\
+        Recursively Looks For Other Topology Files Referred To In The Original File  \\
+        And Adds Them To The List Of Topology Files \\
+        ------------------------------ \\
+        INPUT: \\
+        ---------------  \\
+        topology_input: str -> Name Of Topology \\
+        ------------------------------ \\
+        RETURN: \\
+        ---------------  \\
+        toplist: list -> List Of Topology Files \\
+        ------------------------------ \\
+        '''
+        
+        toplist = []
+        with open(topology_input) as ifile:
+            for line in ifile:
+                match = re.search(r"^;", line, flags=re.MULTILINE)
+                if match:
+                    continue
+                match = re.search(r"^#include\s+\"(\S+)\"", line, flags=re.MULTILINE)
+                if match:
+                    match2 = re.search("ffbonded", match.group(1))
+                    if match2:
+                        continue
+                    match2 = re.search("ffnonbonded", match.group(1))
+                    if match2:
+                        continue
+                    match2 = re.search("forcefield.itp", match.group(1))
+                    if match2:
+                        continue
+                    match2 = re.search("posre.itp", match.group(1))
+                    if match2:
+                        continue
+                    foundname = match.group(1)
+                    check = os.path.isfile(foundname)
+                    if not check:
+                        foundname = os.path.join(self.input_dict['gmxtop_path'], *foundname.strip('/').split('/'))
+                        check = os.path.isfile(foundname)
+                        if not check:
+                            print("File " + foundname + " was not found. Maybe update the gmxpath variable in the script? Exiting.")
+                            exit(1)
+                    toplist.append(foundname)
+
+                    toplist.extend(self.getincludelist(foundname))
+        return toplist
+    
+    def get_atoms(self, qmmm_topology):
```

Comment:
Do we need this function? mass map could be stored in a JSON file for easy read up when necessary, or, if desired, easy change / addition of atom types

Reply:
Do you want to remove the whole function or just exclude the map into a seperate json file? I think we do need the function but putting the map in a json file would definitely improve readability (:

Reply:
I will keep the function and put the map into a json file

Reply:
Done